### PR TITLE
Only run at 7:00 am est (before folks get into the office)

### PR DESCRIPTION
**Pull request by VersusFacit**
### Description

We were running the CI job here 3 times a day which was a massive spike in our Snowflake usage. Let's curtail that to once a night.

Addendum: Try splitting out git jobs so we can run release branches less often without curtailing main testing (at this time).

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR

### CODE REVIEW

File: .github/workflows/release-branch-tests.yml [hunk 1]
```diff
@@ -39,8 +38,8 @@ jobs:
       max-parallel: 1
       fail-fast: false
       matrix:
-        branch: [1.0.latest, 1.1.latest, 1.2.latest, 1.3.latest, main]
-        workflow_name: [main.yml, integration.yml]
+        branch: [1.0.latest, 1.1.latest, 1.2.latest, 1.3.latest]
```
Comment (emmyoop) [inline .github/workflows/release-branch-tests.yml:41]:
```suggestion
        branch: [1.0.latest, 1.1.latest, 1.2.latest, 1.3.latest, 1.4.latest]
```

Comment (emmyoop) [reply .github/workflows/release-branch-tests.yml:41]: Since we cut the 1.4.latest branch yesterday it needs to be added to the matrix.

**Review by emmyoop — APPROVED** [review]
Small change to incorporate but otherwise looks good!

File: .github/workflows/release-branch-tests.yml (hunk 1 above)
Comment (VersusFacit) [reply .github/workflows/release-branch-tests.yml:41]: Oh good callout and yay!

**Pull request merged**